### PR TITLE
Improve star catalog documentation

### PR DESCRIPTION
**Pull request by henrysky**
### Description

I have improved the readability of Star1 structure by put meaningful variable name in Star1 cpp code as well as fixing number error in the star catalog appendix. 

The motivation of this PR is I am experimenting and looking into possible ways to improve star catlogs in Stellarium and want to make little changes without any breaking/new features first.

Fixes # (issue)

N/A

### Screenshots (if appropriate):

N/A

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [x] Housekeeping

### How Has This Been Tested?

I Don't expect there will be any breaking changes nor introduce any new feartures.

**Test Configuration**:
* Operating system: <Name, version number>
* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>

N/A

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [x] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
Did you tested the code on big-endian and little-endian systems?

**Comment by henrysky** [comment]
Only tested on little-endian. Unfortunately I do not have any big-endian computer :(

File: src/core/modules/Star.hpp [hunk 1]
```diff
@@ -96,13 +96,18 @@ struct Star1 { // 28 byte
...
+		quint8  componentIds; // 1 byte
+		qint32  x0;           // 4 bytes
+		qint32  x1;           // 4 bytes
+		quint8  b_v; 		  // 1 byte
+		quint8  vmag;         // 1 bytes
+		quint16 spInt;        // 2 bytes
+		qint32  dx0;          // 4 bytes
+		qint32  dx1;          // 4 bytes
+		qint32  plx;          // 4 bytes
+	} d;  // total is 28 bytes
```
Comment (10110111) [inline src/core/modules/Star.hpp:110]: `static_assert` should be used to make really sure the size is 28 bytes.

File: guide/app_star_catalogue.tex [hunk 2]
```diff
@@ -98,7 +98,7 @@ \subsection{General Description}%\label{general-description}
 \midrule
 \file{stars\_0\_0v0\_8.cat} & 0 & 28 bytes & 0 &      4,979 & Hipparcos\\
 \file{stars\_1\_0v0\_8.cat} & 0 & 28 bytes & 1 &     21,806 & Hipparcos\\
-\file{stars\_2\_0v0\_8.cat} & 0 & 28 bytes & 2 &    150,824 & Hipparcos\\
+\file{stars\_2\_0v0\_8.cat} & 0 & 28 bytes & 2 &    150,826 & Hipparcos\\
```
Comment (10110111) [inline guide/app_star_catalogue.tex:101]: How did you get this value? As I was reading the documentation on the catalog header, I noticed that the magic value given there (`0xde0955a3`) is not what's actually present in the file (`0x835f040a`), and what's defined in `ZoneArray.hpp` in `FILE_MAGIC` (same `0x835f040a`). So I'm not sure if the whole doc there is up to date.

File: src/core/modules/Star.hpp [hunk 3]
```diff
@@ -283,4 +284,4 @@ struct Star3 {  // 6 byte
 };
 static_assert(sizeof(Star3) == 6, "Size of Star3 must be 6 bytes");
 
-#endif // STAR_HPP
+#endif // STAR_HPP
```
Comment (10110111) [inline src/core/modules/Star.hpp:287]: You seem to have removed a trailing `\n` char, please place it back.

**Review by 10110111 — COMMENTED** [review]
The changes to the `struct` make sense to me. I don't think there's any endianness-related problem to be expected, since all that's changed there is the names of the same-sized objects as before.

File: guide/app_star_catalogue.tex (hunk 2 above)
Comment (henrysky) [reply guide/app_star_catalogue.tex:101]: I got the value by parsing the existing catalog. The magic value `0x0a045f83` works for me when I generate new catalog (and everything inside is encoded in little endian). Do you want me to change `0xde0955a3` to `0x0a045f83`?
image

File: src/core/modules/Star.hpp (hunk 1 above)
Comment (henrysky) [reply src/core/modules/Star.hpp:110]: `static_assert` is already in the code

File: src/core/modules/Star.hpp (hunk 3 above)
Comment (henrysky) [reply src/core/modules/Star.hpp]: added it back, sorry about that.

File: src/core/modules/Star.hpp (hunk 1 above)
Comment (10110111) [reply src/core/modules/Star.hpp:110]: Oh, indeed, didn't notice it.

File: guide/app_star_catalogue.tex (hunk 2 above)
Comment (henrysky) [reply guide/app_star_catalogue.tex:101]: Uh you are right, I understand what you mean now. I have updated the guide, other description seems fine to me as far as helping me to parse the existing catalog myself.

**Review by 10110111 — APPROVED** [review]

**Review by alex-w — APPROVED** [review]
Thanks!

@10110111 please merge the PR

File: guide/app_star_catalogue.tex (hunk 2 above)
Comment (gzotti) [reply guide/app_star_catalogue.tex:101]: The star catalog and docs are indeed old. (2006?) The possibly outdated docs on it were collected from the old wiki which was often left behind when something changed. I agree, nowadays the native format should be little-endian, with byte-swapping on the rare big-endians. There is still the proper motion issue to be handled (Guide, F.2 and #3110), I hope @henrysky has this in mind for the next step.

**Pull request merged**